src/v/storage/log_manager.cc

```diff
@@ -854,15 +854,25 @@ ss::future<> log_manager::remove(model::ntp ntp) {
           //
           // TODO: we should more consistently clean up the staging operations
           // to clean up after themselves on failure.
-          if (boost::algorithm::ends_with(de.name, ".staging")) {
+          std::optional<ss::sstring> leftover_file_opt = std::nullopt;
+          if (de.name.ends_with(".staging")) {
               // It isn't necessarily problematic to get here since we can
               // proceed with removal, but it points to a missing cleanup which
               // can be problematic for users, as it needlessly consumes space.
               // Log verbosely to make it easier to catch.
+              leftover_file_opt = "staging";
+          } else if (de.name.ends_with(".cannotrecover")) {
```

Comment:
This is outdated code? Don't see a codepath creating `.cannotrecover` files. In case there is/should be one I suggest naming files with `.ignore_` prefix so that then we can have code that cleanups everything based on prefix rather than maintaining special cases.